File: backend/src/merchant/objects/apiTypes/MerchantMeResponse.scala
package delivery.merchant.objects.apiTypes

import delivery.admin.objects.StoreOnboardingRequest
import delivery.merchant.objects.MerchantAccountPublic
import delivery.shared.objects.UserRole

final case class MerchantMeResponse(
    username: String,
    role: UserRole,
    merchantAccount: MerchantAccountPublic,
    onboardingRequests: List[StoreOnboardingRequest] = Nil
)

File: frontend/src/objects/merchant/apiTypes/MerchantMeResponse.ts
import { UserRoles } from '@/objects/shared/ids'
import type { UserRole } from '@/objects/shared/ids'
import type { StoreOnboardingRequest } from '@/objects/admin/StoreOnboardingRequest'
import type { MerchantAccountPublic } from '../MerchantAccountPublic'

export interface MerchantMeResponse {
  username: string
  role: typeof UserRoles.merchant & UserRole
  merchantAccount: MerchantAccountPublic
  onboardingRequests?: StoreOnboardingRequest[]
}
